### PR TITLE
Add sign out btn

### PR DESCRIPTION
**Pull request by denar90**
- sign out btn was added
- service worker registartion runs after viewer instance was created. I faced this issue during opening TV with cleaned cache.

### CODE REVIEW

**Comment by denar90** [comment]
btw, ignore travis error. it's just our one cypress test

File: docs/auth.js [hunk 1]
```diff
@@ -45,9 +45,7 @@ class GoogleAuth {
     return GoogleAuth.getAuthInstance().signIn(oAuthOptions);
   }
   signOut() {
-    try {
-      this.getAuthInstance().signOut();
```
Comment (paulirish) [inline docs/auth.js]: hah. nice silent error we had there. :)

good fix.

File: docs/index.html [hunk 2]
```diff
@@ -109,6 +110,24 @@ <h3 class="f4 br3 br--top black-60 pv2 mb0">Wanna compare traces?</h3>
 
         // We are monkeypatching window.loadResourcePromise, which is from devtools' Runtime.js
         viewer.devTools.monkeypatchLoadResourcePromise();
+
+        if ('serviceWorker' in navigator) {
```
Comment (denar90) [inline docs/index.html]: yup 👍

**Comment by denar90** [comment]
oh yeah, about sign out you are right. looks like it will sign out from google acct. I'll take a look at docs to figure this out.

File: docs/index.html [hunk 3]
```diff
@@ -79,7 +80,19 @@ <h3 class="f4 br3 br--top black-60 pv2 mb0">Wanna compare traces?</h3>
...
+          return;
+        }
+        const viewer = new Viewer();
+
+        viewer.checkAuth({immediate: true});
+
+        // We are monkeypatching window.loadResourcePromise, which is from devtools' Runtime.js
+        viewer.devTools.monkeypatchLoadResourcePromise();
+
         navigator.serviceWorker
```
Comment (denar90) [inline docs/index.html]: We event don't need  `if ('serviceWorker' in navigator)`

Comment (paulirish) [inline docs/index.html]: true, especially we have the `if (!window.chrome)` above this.
sgtm

**Comment by paulirish** [comment]
last question.. what is motivating the signout button? 

if we don't need it, i'd prefer to not show it.

**Comment by denar90** [comment]
For me, it's easy to debug. I'm playing with different accounts, also when I signed in and signed out.
From other point, users might want it. The case, when they can't sign out from a place where they signed in, scares 👻 So +1 for good ux I think. 
wdyt, why not to have it?

**Comment by paulirish** [comment]
> why  not to have it?

basically because it adds more UI that may be unnecessary.
i think for testing we can make great use of incognito
but i can't think of cases where a user would want to use our signout button to sign out of their google account.

Alternatively, i could see the user wanting a one-click to remove the Drive permissions they granted to this app. (we can't do that without asking for MORE permissions, but we could more prominently link them)

**Comment by denar90** [comment]
>i think for testing we can make great use of incognito

yeah, make sense :)

----
>but i can't think of cases where a user would want to use our signout button to sign out of their google account.

I've changed it https://github.com/ChromeDevTools/timeline-viewer/pull/33/commits/f747fea4c6d06a53796547e674d8add9307da7a3, now it's disconnecting or app due to https://developers.google.com/identity/sign-in/web/disconnect

So it should solve
>Alternatively, i could see the user wanting a one-click to remove the Drive permissions they granted to this app. (we can't do that without asking for MORE permissions, but we could more prominently link them)

**Comment by denar90** [comment]
maybe rename function `signOut` -> `revokeAccess`

**Comment by paulirish** [comment]
ya let's rename the function and the button.

"Revoke API Access"

sounds awesome.

**Pull request merged**